repairnator/repairnator-scanner/src/main/java/fr/inria/spirals/repairnator/scanner/ScannerBuildListener.java

```diff
@@ -1,26 +1,100 @@
 package fr.inria.spirals.repairnator.scanner;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import fr.inria.spirals.repairnator.config.RepairnatorConfig;
 import fr.inria.spirals.repairnator.Listener;
-import javax.jms.MessageListener;
+
 import javax.jms.Message;
+import javax.jms.Connection;
+import javax.jms.Destination;
+import javax.jms.JMSException;
+import javax.jms.Session;
+import javax.jms.TextMessage;
+import javax.jms.BytesMessage;
+import javax.jms.ConnectionFactory;
+import javax.jms.MessageListener;
+import javax.jms.MessageProducer;
+import javax.jms.MessageConsumer;
 
+import org.apache.activemq.ActiveMQConnectionFactory;
 /**
  * This class fetch build ids from ActiveMQ queue and run the pipeline with it.
  */
 public class ScannerBuildListener implements Listener,MessageListener{
+    private static final Logger LOGGER = LoggerFactory.getLogger(ScannerBuildListener.class);
+    private static final RepairnatorConfig config = RepairnatorConfig.getInstance();
     private static Launcher launcher;
 
     public ScannerBuildListener (Launcher launcher) {
         this.launcher = launcher;
     }
     
-    public void runListenerServer() {}
+    public void submitBuild(String buildStr) {
```

Comment:
This method could be declared in interface `Listener`, so that you will avoid type casting in your code: `ScannerBuildListener scannerBuildListener = (ScannerBuildListener) listener;`